### ISSUE TITLE
Data -> Validity -> List

### ISSUE THREAD

**Issue by xinstein** [CollaboraOnline/online#6956]
**Describe the bug**
When I enter multiple lines in the `Entries` box (Which I assume is the way to separate list entries), I get the following error in collabora log and the operation fails. 
```
collabora_app_1        | kit-00132-00017 2023-07-22 03:31:29.972470 +0000 [ kitbroker_008 ] ERR  ToMaster-0be: Exception while handling [dialogevent 77 {"id":"minlist", "cmd": "change", "data": "fsdfwqe]: <unspecified file>(1): unterminated string| common/Session.cpp:277
```

**To Reproduce**
1. Go to 'Data -> Validity'
2. Select on 'List'
3. Enter multi-line string in `Entries`

**Expected behavior**
It should work

**Actual behavior**
It fires an error which is clearly a bug.

**Screenshots**

**Desktop (please complete the following information)**
 - OS: Windows 10
 - Browser: Chromium
 - Version: 113

**Additional context**
running collabora 23.05, on nextcloud 25

**Comment by xinstein** [comment]
The workaround is to use \n instead of newline

**Comment by timur-g** [comment]
I did not understand what you wanted to achieve, works OK for me.
![image](https://github.com/CollaboraOnline/online/assets/7240617/26b7c6ee-3f06-4e7b-941f-e335ccc7d450)

Are you trying multiple lines with some of them being multi-line strings? Please give example, best as ODS.

**Comment by xinstein** [comment]
![image](https://github.com/CollaboraOnline/online/assets/10842079/092e0d01-f844-4f8b-9ed8-bf40efa1b618)
I'm trying to enter allowed options here. I found no document explaining how to separate options.
So I tried using `;`, `,`, as well as return and `\n` as seperator, none of those works except \n

**Comment by xinstein** [comment]
> ![image](https://user-images.githubusercontent.com/10842079/256201405-092e0d01-f844-4f8b-9ed8-bf40efa1b618.png) I'm trying to enter allowed options here. I found no document explaining how to separate options. So I tried using `;`, `,`, as well as return and `\n` as seperator, none of those works except \n

As in the screenshot, if I hit `OK` it will cause the aforementioned error.
I have to change the text to `foo\nbar\nbaz` in order to meet my purpose

**Comment by timur-g** [comment]
So this does not seem like a bug, but missing documentation, I confirm as such.

**Comment by xinstein** [comment]
> So this does not seem like a bug, but missing documentation, I confirm as such.

Is it intended that the `Entries` textbox does not accept newline character?

But I clearly remember entering newline in this box last year without triggering any error, now I have to use `\n` for every newline.

Worse yet, the next time I open this setting (to  maybe edit the options), all the `\n` becomes newline automatically, and I have to change them back to \n one by one, I don't believe this is intended behaviour.

**Comment by xinstein** [comment]
Also, if you read the error message in my original post, you'll see that it is caused by reading an incomplete json string (the json ends with `"fsdfwqe` abruptly)

**Comment by eszkadev** [comment]
If this produces error -> should be fixed, and it is a bug.
If I understand correctly typing `\n` characters works but proper newline (using enter) not? This also should be fixed, it's a problem with encoding/decoding here.

**Comment by xinstein** [comment]
> If I understand correctly typing `\n` characters works but proper newline (using enter) not?

Exactly.

I'm using nginx as reverse proxy.

I've reproduced the error on Microsoft Edge and chromium (Linux and Windows).

**Comment by xinstein** [comment]
> As shown in screenshot, I see that entering new line with enter works, so that does not seem like a common bug.

would you please read the text as well as the screenshot?

**Comment by vanantwerp** [comment]
I am also experiencing this issue, but in addition, the `\n` workaround does NOT work for me.

**Comment by lvillanne-recia** [comment]
Hello,
I have the same problem.
I've looked a bit further to try and find the source of the bug.

I can provide the logs of a simple example where I type the following text in the field :
```
12
3
```
This is what is sent to the websocket :
```
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "1", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12
", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12
3", "type": "edit"}
```

And here's what I get from the server-side logs:
```
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "1", "type": "edit"} | kit/Kit.cpp:2561
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12", "type": "edit"} | kit/Kit.cpp:2561
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12 | kit/Kit.cpp:2561
ERR  ToMaster-1a9f: Exception while handling [dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12]: <unspecified file>(1): unterminated string| common/Session.cpp:277
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12 | kit/Kit.cpp:2561
ERR  ToMaster-1a9f: Exception while handling [dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12]: <unspecified file>(1): unterminated string| common/Session.cpp:277
```

We can see that as soon as the field content has a line break, we get an incomplete string which is cut off at the line break.
What's more, this systematically triggers the following error:
```
ERR  ToMaster-1a9f: Exception while handling [....]: <unspecified file>(1): unterminated string| common/Session.cpp:277
```


If I try replacing the carriage return with a `\n`, it works:
```
12\n3
```

In this case, this is what is sent to the websocket :
```
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "1", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\n", "type": "edit"}
dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\n3", "type": "edit"}
```
And this is what I get back in the server-side logs:
```
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "1", "type": "edit"} | kit/Kit.cpp:2561
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12", "type": "edit"} | kit/Kit.cpp:2561
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"} | kit/Kit.cpp:2561
ERR  ToMaster-1a9f: Exception while handling [dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"}]: <unspecified file>(1): expected '}' or ','| common/Session.cpp:277
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"} | kit/Kit.cpp:2561
ERR  ToMaster-1a9f: Exception while handling [dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\", "type": "edit"}]: <unspecified file>(1): expected '}' or ','| common/Session.cpp:277
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\n", "type": "edit"} | kit/Kit.cpp:2561
DBG  #19: child_ws: recv [child-1a9f dialogevent 5 {"id":"minlist", "cmd": "change", "data": "12\n3", "type": "edit"} | kit/Kit.cpp:2561
```

There's always an error that highlights another bug. The `\` which are alone are not escaped and consequently the string is no longer valid because the `\` has escaped the `"` at the end of the `data` string.

**Comment by lvillanne-recia** [comment]
The bug seems to be present in all multiline text fields.
For example, in the comments field of a document (`File/Properties` then `Description` tab and `Comments` field).
You can therefore reproduce the bug when editing an ods, but also an odt in the same way.

On the other hand, if I compare it with something that works, for example when editing an odt, adding a comment (`Insert/Comment`) with the following comment :
```
12
3
```
I get the following json in the websocket:
```
uno .uno:InsertAnnotation {"Text":{"type":"string","value":"12\n3"},"Author":{"type":"string","value":"******"}}
```

You can see here that the line break has been automatically replaced by a `\n` in the json.
So I think the bug comes from the fact that in multiline text fields the line break is not substituted by a `\n` on the javascript side.

**Comment by timur-g** [comment]
Fix will be in 23.05.7.
